### PR TITLE
SK-1277: Choose payment method for on-chain DID

### PR DESCRIPTION
**Pull request by lukeg90**
SK-1277
SK-1411

### CODE REVIEW

**Comment by lukeg90** [comment]
Not to be merged until checkout app is implemented

File: src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx [hunk 1]
```diff
@@ -74,12 +105,79 @@ export function DidUpgradeExplainer({ identity }: Props): JSX.Element | null {
...
+                : styles.paymentMethods
+            }
+          >
+            <p className={styles.paymentMethod}>
+              <label
+                className={
+                  insufficientKilt ? styles.insufficientKilt : styles.kilt
+                }
+              >
+                <input
```
Comment (arty-name) [inline src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx:121]: Since we now have inputs here, would it make sense to make the outermost tag a `form`?

**Review by arty-name — APPROVED** [review]

File: src/views/DidUpgradeEuro/DidUpgradeEuro.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+interface Props {
+  identity: Identity;
+}
+
+// TODO: Fetch submitter address from TXD
+const submitter = '4t37z5PrEH9zz93cQ2of8F9kYMPrmWcRMBckJtNGF8keSW5W';
+
+// TODO: return TXD URL when it works to fetch submitter address
+async function getExternalURLs(): Promise<{ checkoutBaseURL: string }> {
```
Comment (arty-name) [inline src/views/DidUpgradeEuro/DidUpgradeEuro.tsx:31]: Would it be nice to have it as a mapping, like [this one](https://github.com/BTE-Trusted-Entity/sporran-extension/blob/main/src/utilities/useSubscanHost/useSubscanHost.ts)?

File: src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx [hunk 3]
```diff
@@ -26,12 +37,35 @@ export function DidUpgradeExplainer({ identity }: Props): JSX.Element | null {
 
   const wasOnChainDidDeleted = useIsOnChainDidDeleted(did);
 
+  const { total: kiltCosts, insufficientKilt } = useKiltCosts(address, did);
+
+  const [paymentMethod, setPaymentMethod] = useState<'kilt' | 'euro'>('kilt');
```
Comment (arty-name) [inline src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx:42]: Having this enumeration in two places already, would make sense to give it a name

File: src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx [hunk 4]
```diff
@@ -74,12 +108,82 @@ export function DidUpgradeExplainer({ identity }: Props): JSX.Element | null {
             {t('view_DidUpgradeExplainer_deposit')}
           </p>
 
+          {/* TODO: remove feature check when KILT checkout is released */}
```
Comment (arty-name) [inline src/views/DidUpgradeExplainer/DidUpgradeExplainer.tsx:111]: Is this comment superfluous? When removing the feature, we’ll be forced to clean this up…

**Review by arty-name — APPROVED** [review]

**Pull request merged**